app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/SettingsActivity.java

```diff
@@ -261,18 +274,59 @@ public boolean onPreferenceClick(Preference preference) {
         });
 
         pref = findPreference("weather_city");
+        pref.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
+           @Override
+           public boolean onPreferenceChange(Preference preference, Object newVal) {
+               // reset city id and force a new lookup
+               GBApplication.getPrefs().getPreferences().edit().putString("weather_cityid", null).apply();
+               preference.setSummary(newVal.toString());
+               Intent intent = new Intent("GB_UPDATE_WEATHER");
+               intent.setPackage(BuildConfig.APPLICATION_ID);
+               sendBroadcast(intent);
+               return true;
+           }
+        });
+
+        pref = findPreference(GBPrefs.AUTO_EXPORT_LOCATION);
+        pref.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+            public boolean onPreferenceClick(Preference preference) {
+                Intent i = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+                i.setType("application/x-sqlite3");
+                i.addCategory(Intent.CATEGORY_OPENABLE);
+                i.addFlags(Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION | Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
+                String title = getApplicationContext().getString(R.string.choose_auto_export_location);
+                startActivityForResult(Intent.createChooser(i, title), FILE_REQUEST_CODE);
+                return true;
+            }
+        });
+        pref.setSummary(getAutoExportLocationSummary());
+
+        pref = findPreference(GBPrefs.AUTO_EXPORT_INTERVAL);
         pref.setOnPreferenceChangeListener(new Preference.OnPreferenceChangeListener() {
             @Override
-            public boolean onPreferenceChange(Preference preference, Object newVal) {
-                // reset city id and force a new lookup
-                GBApplication.getPrefs().getPreferences().edit().putString("weather_cityid",null).apply();
-                preference.setSummary(newVal.toString());
-                Intent intent = new Intent("GB_UPDATE_WEATHER");
-                intent.setPackage(BuildConfig.APPLICATION_ID);
-                sendBroadcast(intent);
+            public boolean onPreferenceChange(Preference preference, Object autoExportInterval) {
+                String summary = String.format(
```

Comment:
JFYI (no need to change this): getString() is overloaded with a var-args Object parameters, so you can directly add arguments to the String without the need to use String.format(). The placeholder format is a bit different though (e.g. `%1$s` for the first string argument).